root/usr/bin/nm-prepare

```diff
@@ -4,21 +4,23 @@ echo "Kernel command line: $(cat /proc/cmdline)"
 bootif=$(cat /proc/cmdline | grep -io 'BOOTIF=\S*' | tail -n1 | tr A-F- a-f: | cut -d: -f2-7)
 echo "Parsed booting interface: $bootif"
 
-for i in /sys/class/net/*; do
-  dev=$(basename $i)
-  [[ "$dev" == "lo" ]] && continue
-  mac=$(cat /sys/class/net/$dev/address | tr A-F- a-f:)
-  if [[ "$mac" == "$bootif" ]]; then
-    echo "Configuring interface $i (default route/DNS)"
-    peerdns=yes
-    defroute=yes
-  else
-    echo "Configuring interface $i"
-    peerdns=no
-    defroute=no
-  fi
-  script=/etc/sysconfig/network-scripts/ifcfg-$dev
-  cat > $script <<EONS
+dev="$1"
+
+[ "$dev" == "lo" ] && exit
+
+read mac < "/sys/class/net/${dev}/address"
+
+if [[ "$mac" == "$bootif" ]]; then
+  echo "Configuring interface ${dev} (default route/DNS)"
+  peerdns=yes
+  defroute=yes
+else
+  echo "Configuring interface ${dev}"
+  peerdns=no
+  defroute=no
+fi
+script=/etc/sysconfig/network-scripts/ifcfg-$dev
+cat > $script <<EONS
 DEVICE=$dev
 TYPE=Ethernet
 BOOTPROTO=dhcp
```

Comment:
Damn, this whole file has been completely rewritten in d4cb4ea7f4f84396542ebd1e9c905e032e11cedd and it also triggers via udev now. You can basically drop your changes as it works the same way (accepts a parameter = device name). It now does not create rh-cfg plugin configurations but directly NetworkManager ini-style profiles instead. I also use `systemd-cat` utility to redirect output to journal.